packages/connections/src/utils/validation-errors.spec.ts

```diff
@@ -0,0 +1,284 @@
+      expect(result).to.be.empty;
+    });
+
+    it('should not return error if ssl is enabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=true&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.be.empty;
+    });
+
+    it('should return error if no certificate is provided', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=true`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'A Client Certificate is required with x509 authentication.',
+        },
+      ]);
+    });
+  });
+  describe('LDAP', function () {
+    it('should return errors if no username is defined', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://:password@myserver.com?authMechanism=PLAIN`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'Username and password are required.',
+        },
+      ]);
+    });
+    it('should return errors if no password is defined', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://username:@myserver.com?authMechanism=PLAIN`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'Username and password are required.',
+        },
+      ]);
+    });
+  });
+  describe('Kerberos', function () {
+    it('should return errors if no principal is defined when using Kerberos', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=GSSAPI`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          field: 'kerberosPrincipal',
+          message: 'Principal name is required with Kerberos.',
+        },
+      ]);
+    });
+    it('should not return errors if principal is defined when using Kerberos', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:myService`,
```

Comment:
the principal is the username:

```suggestion
          connectionString: `mongodb://pricipal@myserver.com?authMechanism=GSSAPI`,
```